source/states/BaseState.hx

```diff
@@ -0,0 +1,96 @@
+package states;
+
+import flixel.FlxObject;
+
+/**
+ * General state class for representing states for any generic FlxObject
+ *
+ * As a note, a state should be responsible for calling its nextState's enter
+ * method, and also responsible for calling its own exit method
+ *
+ * @author Garren Ijames
+ */
+class BaseState
+{
+  public  var nextState:Null<BaseState>;
+
+  private var duration:Int;
```

Comment:
I don't think that would be difficult to implement. With proper naming I don't think that would be too confusing either. There may be certain states that need to have a measure of "framesLeft" as well as a time based duration value as well. I will make it happen